Makefile

```diff
@@ -585,8 +587,18 @@ rpm: tarball test
 
 # Removing test dependency until resolved
 tarball:
-	VERSION=`awk '/^Version/ {print $$2}' deepsea.spec`; \
-	git archive --prefix deepsea-$$VERSION/ -o ~/rpmbuild/SOURCES/deepsea-$$VERSION.tar.bz2 HEAD
+	mkdir -p .tmp
+	git archive --prefix deepsea-tmp/ -o .tmp/deepsea-tmp.tar HEAD
+	HASH=`git rev-parse --short HEAD`; \
+	OFFSET=`git rev-list \`git log --pretty=oneline -- version.txt | head -1 | cut -d ' ' -f 1\`..HEAD | wc -l`; \
+	cd .tmp; tar -xf deepsea-tmp.tar; cd deepsea-tmp; \
+	VERSION=`cat version.txt`; F_VERSION="$$VERSION+git.$$OFFSET.$$HASH"; \
```

Comment:
In ceph the RPM version number format is `x.y.z-offset-sha1` (e.g. 13.0.0-3504-ga292d52fe7) and for this I have verified that RPM considers e.g. the following series to be ascending:

1. `13.0.0-9-g$SHA1`
2. `13.0.0-10-g$SHA1`

However, I'm not sure if RPM will be so savvy with this series:

1. `13.0.0+git.9.$SHA1`
2. `13.0.0+git.10.$SHA1`

That will need testing before this PR is merged?

Reply:
I personally am not a big fan of adding git revision hashes to version numbers, as they are random and don't really indicate any ordering. A timestamp is much more readable IMHO - the corresponding git revision hash could still be stored in the version.txt file, of course.

Reply:
@LenzGr Having the git hash/SHA1 in the version number is useful for developers because it enables easy cross-linking between "code in RPM" and "code in git". However, it's useless for ensuring that the RPM version number is always increasing. That's why the timestamp is sometimes used, but git offset is another option.

It's interesting to note here that the OBS replaces whatever is in `Release: ...` in the spec file with its own, always ascending, number (the OBS revision number). So, as long as we populate the `Version: ...` line in the spec file with *just* the major version number (e.g. `0.8.0`) and put the other components (offset, sha1, timestamp, etc.) in `Release: ...`, it doesn't matter because OBS ensures that each successive RPM version number will be greater than the one before.

Reply:
For an example of how this works, see https://build.opensuse.org/package/show/filesystems:ceph:mimic/ceph -- in the spec file, the "Release:" line says "3535.g8b10d08c04" (git offset and hash), but the built RPMs look like this: `ceph-13.0.0-194.1.src.rpm` because, during the build, OBS does this:

```
[  113s] -----------------------------------------------------------------
[  113s] I have the following modifications for ceph.spec:
[  113s] 71c71
[  113s] < Release:	3535.g8b10d08c04%{?dist}
[  113s] ---
[  113s] > Release:	194.1
[  113s] -----------------------------------------------------------------
```

Reply:
> However, I'm not sure if RPM will be so savvy with this series:
>
>    1. 13.0.0+git.9.$SHA1
>    2. 13.0.0+git.10.$SHA1

This is the deepsea version string in the package name. Has worked fine so far. I'll double check.